### ISSUE TITLE
IDISP012 should not warn when invoking unknown func.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#72]
```cs
namespace RoslynSandbox
{
    using System;

    public sealed class Foo
    {
        private readonly Func<IDisposable> func;

        public Foo(Func<IDisposable> func)
        {
            this.func = func;
        }

        public IDisposable Disposable => this.func();
    }
}
```
